module/platform/OpenCR/HardwareIO/src/hardwareio/handle_response.cpp

```diff
@@ -167,12 +177,59 @@ namespace module::platform::OpenCR {
                         if (response.size() == size_t(7 + packet_length)) {
                             current_phase = Phases::FINISH;
                         }
+                        // Go to UNSTUFF_1 if we see 0xFF in the data field and we haven't finished yet
+                        else if (buf[i] == packet_header[0]) {
+                            current_phase = Phases::UNSTUFF_1;
+                        }
                     }
                     else {
                         current_phase = Phases::TIMEOUT;
                     }
                     break;
 
+                // Going to the case below means that we saw a 0xFF in the data field
+                // If the next byte is another 0xFF, then we go to UNSTUFF_2
+                case Phases::UNSTUFF_1:
+                    current_phase = buf[i] == packet_header[1] ? Phases::UNSTUFF_2 : Phases::DATA;
```

Comment:
I think we might need the timeout check here (and in other unstuffing phases)

Reply:
I revised the fsm so we dont have to write as much code. Now we only have to check for timeouts in one data case instead of many like in here